src/sysdolphin/baselib/jobj.c

```diff
@@ -23,3 +27,250 @@ void HSD_JObjCheckDepend(HSD_JObj* jobj)
+        cb(jobj, args, type);
+    }
+    if (!(jobj->flags & 0x1000)) {
+        HSD_JObj* child = jobj->child;
+        while (child != NULL) {
+            func_8036EFAC(child, cb, args);
+            child = child->next;
+        }
+    }
+}
+
+void func_8036F0F0(HSD_JObj* jobj, void (*cb)(HSD_JObj*, void*, u32), void* args)
+{
+    if (jobj == NULL) {
+        return;
+    }
+    if (cb != NULL) {
+        cb(jobj, args, 0);
+    }
+    if (!(jobj->flags & 0x1000)) {
+        HSD_JObj* child = jobj->child;
+        while (child != NULL) {
+            func_8036EFAC(child, cb, args);
+            child = child->next;
+        }
+    }
+}
+
+inline BOOL has_scl(HSD_JObj* jobj)
+{
+    BOOL result = FALSE;
+    if (jobj != NULL && jobj->scl != NULL)
+        result = TRUE;
+    return result;
+}
+
+void HSD_JObjMakeMatrix(HSD_JObj* jobj)
+{
+    Vec* scl;
+
+    HSD_JObjSetupMatrix(jobj->parent);
+    if (jobj->flags & 8) {
+        if (jobj->parent != NULL && jobj->parent->scl != NULL) {
+            if (jobj->scl == NULL) {
+                jobj->scl = HSD_VecAlloc();
+            }
+            *jobj->scl = *jobj->parent->scl;
+        } else {
+            if (jobj->scl != NULL) {
+                HSD_VecFree(jobj->scl);
+                jobj->scl = NULL;
+            }
+        }
+    } else {
+        if (jobj->scl == NULL) {
+            jobj->scl = HSD_VecAlloc();
+        }
+        if (jobj->parent != NULL && jobj->parent->scl != NULL) {
+            jobj->scl->x = jobj->scale.x * jobj->parent->scl->x;
+            jobj->scl->y = jobj->scale.y * jobj->parent->scl->y;
+            jobj->scl->z = jobj->scale.z * jobj->parent->scl->z;
+        } else {
+            *jobj->scl = jobj->scale;
+        }
+    }
+    if (jobj->flags & 0x20000) {
+        if (has_scl(jobj->parent)) {
+            scl = jobj->parent->scl;
+        } else {
+            scl = NULL;
+        }
+        HSD_MtxSRTQuat(jobj->mtx, &jobj->scale, &jobj->rotate, &jobj->translate, scl);
+    } else {
+        if (has_scl(jobj->parent)) {
+            scl = jobj->parent->scl;
+        } else {
+            scl = NULL;
+        }
+        HSD_MtxSRT(jobj->mtx, &jobj->scale, (Vec*) &jobj->rotate, &jobj->translate, scl);
+    }
+    if (jobj->parent != NULL) {
+        PSMTXConcat(jobj->parent->mtx, jobj->mtx, jobj->mtx);
+    }
+    if (jobj->aobj != NULL && jobj->aobj->hsd_obj != NULL) {
+        Vec vec;
+        HSD_JObj* aobj_jobj = (HSD_JObj*) jobj->aobj->hsd_obj;
+        HSD_JObjSetupMatrix((HSD_JObj*) jobj->aobj->hsd_obj);
+        PSMTXMUltiVec(aobj_jobj->mtx, &jobj->translate, &vec);
+        jobj->mtx[0][3] = vec.x;
+        jobj->mtx[1][3] = vec.y;
+        jobj->mtx[2][3] = vec.z;
+    }
+}
+
+void HSD_JObjRemoveAnimByFlags(HSD_JObj* jobj, u32 flags)
+{
+    BOOL has_dobj;
+    if (jobj != NULL) {
+        if (flags & 1) {
+            HSD_AObjRemove(jobj->aobj);
+            jobj->aobj = NULL;
+        }
+        if (jobj->flags & 0x4020) {
```

Comment:
This looks like `union_type_dobj` or some type of inline, given it shows up multiple times.